dask/bag/core.py

```diff
@@ -619,7 +619,10 @@ def map_partitions(self, func, *args, **kwargs):
         *args, **kwargs : Bag, Item, Delayed, or object
             Arguments and keyword arguments to pass to ``func``.
             Partitions from this bag will be the first argument, and these will
-            be passed *after*.
+            be passed *after*. Depending on previous tasks, dask objects may be
+            passed to ``func`` as either python generators or lists. You may
+            call ``list`` from within ``func`` or explicitly call
+            ``bag.map_partitions(list)`` beforehand.
```

Comment:
I think that we should avoid listing ways to turn iterators into a list here.  This isn't necessarily going to be the problem that the user will face.  Instead I think we should be explicit about what the user's function should expect.  I think in this case it's that functions should expect to receive either `Iterator`s or `Iterable`s, depending on the preceding tasks in the task graph.